iConverter/DI/MainAssembly.swift

```diff
@@ -10,26 +10,37 @@ import SwinjectAutoregistration
 
 final class MainAssembly: Assembly {
     func assemble(container: Container) {
-        
-        container.autoregister(
-            MainViewModel.self,
-            initializer: MainViewModel.init
-        )
-        
+
+        // MARK: - Home
+        container.autoregister(MainRouter.self, initializer: MainRouter.init).inObjectScope(.container)
+        container.autoregister(MainInteractor.self, initializer: MainInteractor.init)
+        container.autoregister(MainPresenter.self, initializer: MainPresenter.init)
+        container.autoregister(MainViewModel.self, initializer: MainViewModel.init)
         container.register(MainViewController.self) { r in
             let controller = MainViewController.instantiate()
-            controller.viewModel = r ~> MainViewModel.self
-            
+            controller.makeDI(
```

Comment:
Please remove dependency to nib (use Stevia) so you can inject this via constructor, we won't mutate these anyway.